tests/unit/OAuth/OAuth/MiddlewareTest.php

```diff
@@ -0,0 +1,321 @@
+<?php
+
+namespace Penneo\SDK\Tests\Unit\OAuth\OAuth;
+
+use BlastCloud\Guzzler\UsesGuzzler;
+use Carbon\Carbon;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\BadResponseException;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use Penneo\SDK\AuthenticationExpiredException;
+use Penneo\SDK\OAuth\Tokens\PenneoTokens;
+use Penneo\SDK\OAuth\Tokens\SessionTokenStorage;
+use Penneo\SDK\PenneoSDKException;
+use Penneo\SDK\Tests\Unit\OAuth\BuildsOAuth;
+use PHPUnit\Framework\TestCase;
+
+class MiddlewareTest extends TestCase
+{
+    use BuildsOAuth;
+    use UsesGuzzler;
+
+    private $tomorrowTimestamp;
+    private $yesterdayTimestamp;
+    private $fiveSecondsInTheFutureTimestamp;
+
+    /**
+     * @var SessionTokenStorage
+     */
+    private $mockStorage;
+
+    public function setUp(): void
+    {
+        Carbon::setTestNow(Carbon::create(1998, 5, 10, 5, 10, 36));
```

Comment:
Is this date important or could we just use `::now()` instead?

Reply:
It is not! I've replaced it with `::now()` for the sake of simplicity.